wit/types.wit

```diff
@@ -36,40 +36,69 @@ interface types {
     unexpected-error(string)
   }
 
-  /// This following block defines the `fields` resource which corresponds to
-  /// HTTP standard Fields.
-  resource fields {
-    /// Multiple values for a header are multiple entries in the list with the
-    /// same key.
-    constructor(entries: list<tuple<string,list<u8>>>);
-
-    /// Values off wire are not necessarily well formed, so they are given by
-    /// list<u8> instead of string.
-    get: func(name: string) -> list<list<u8>>;
+  /// Field keys are always strings.
+  type field-key = string;
 
-    /// Values off wire are not necessarily well formed, so they are given by
-    /// list<u8> instead of string.
-    set: func(name: string, value: list<list<u8>>);
-    delete: func(name: string);
-    append: func(name: string, value: list<u8>);
+  /// Field values should always be UTF-8 encoded strings. However, in
```

Comment:
```suggestion
  /// Field values should always be ASCII strings. However, in
```
It [sounds](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.4) like anything outside ASCII is sketchy.